tests/unit/test_collector.py

```diff
@@ -218,9 +282,18 @@ def test_determine_base_url(html, url, expected):
         # not. The `:` should be quoted.
         ("https://localhost.localdomain/T:/path/",
          "https://localhost.localdomain/T%3A/path/"),
+        # URL with a quoted "/" in the path portion.
+        ("https://example.com/access%2Ftoken/path/",
+         "https://example.com/access%2Ftoken/path/"),
         # VCS URL containing revision string.
         ("git+ssh://example.com/path to/repo.git@1.0#egg=my-package-1.0",
          "git+ssh://example.com/path%20to/repo.git@1.0#egg=my-package-1.0"),
+        # VCS URL with a quoted "#" in the revision string.
+        ("git+https://example.com/repo.git@hash%23symbol#egg=my-package-1.0",
+         "git+https://example.com/repo.git@hash%23symbol#egg=my-package-1.0"),
+        # VCS URL with a quoted "@" in the revision string.
+        ("git+https://example.com/repo.git@at%40 space#egg=my-package-1.0",
+         "git+https://example.com/repo.git@at%40%20space#egg=my-package-1.0"),
```

Comment:
Perhaps add a test with "&subdirectory=some relative path" ?

Reply:
This patch only affects the path part or the URL, not the query or hash, so the `subdirectory` part is not covered.

It is likely a good idea to add a test case for it anyway (I casually searched and couldn’t find any), but should be done in a separate issue/PR IMO.

Reply:
I filed #7605 as a follow-up.